generics_other.go

```diff
@@ -0,0 +1,12 @@
+//go:build !go1.18
+// +build !go1.18
+
+package swag
+
+func typeSpecFullName(typeSpecDef *TypeSpecDef) string {
+	return typeSpecDef.FullName()
+}
+
+func (pkgDefs *PackagesDefinitions) parametrizeStruct(original *TypeSpecDef, fullGenericForm string) *TypeSpecDef {
```

Comment:
How do you expect to pass the tests for older go versions if this is not implemented?

Reply:
I guess you will have to put the TestParseGenericsBasic into a separate test file and add build tags.

The `typeSpecDef.TypeSpec.TypeParams` used in packages.go isn't available in < 1.18, so will have to use the same approach as in 
https://github.com/golang/tools/blob/master/internal/typeparams/typeparams_go118.go
https://github.com/golang/tools/blob/master/internal/typeparams/typeparams_go117.go

Reply:
Or use https://pkg.go.dev/golang.org/x/exp/typeparams@v0.0.0-20220428152302-39d4317da171

Reply:
The swag generator must produce the same output no matter if it is built with go1.6 or go1.18.

Reply:
@NicklasWallgren Thanks for your suggestions. I will give it a try with typeparams.

Reply:
@kirillDanshin any update here?